File: bisection_search/bisection_search.py
def isIn(char, aStr):
    '''
    char: a single character
    aStr: an alphabetized string

    returns: True if char is in aStr; False otherwise
    '''
    aStr = ''.join(sorted(aStr))
    length = len(aStr)
    mid = length // 2
    if aStr == '':
        return False
    elif len(aStr) == 1:
        return aStr == char
    elif char == aStr[mid]:
        return True
    elif char < aStr[mid]:
        return isIn(char, aStr[:mid])
    else:
        return isIn(char, aStr[mid+1:])

File: bisection_search/bisection_search.rb
# char: a single character
# a_str: an alphabetized string
# prints: true if char is in a_str; false otherwise

def is_in(char, a_str)

  a_str = a_str.chars.sort(&:casecmp).join
  length = a_str.length
  mid = length / 2
  if a_str == ''
      p false
  elsif a_str.length == 1
      a_str == char
  elsif char == a_str[mid]
      p true
  elsif char < a_str[mid]
      is_in(char, a_str[0..(mid-1)])
  else
      is_in(char, a_str[(mid+1)..-1])
  end
end
